File: ios/Classes/BackgroundDownloaderPlugin.m
#import "BackgroundDownloaderPlugin.h"
#if __has_include(<background_downloader/background_downloader-Swift.h>)
#import <background_downloader/background_downloader-Swift.h>
#else
// Support project import fallback if the generated compatibility header
// is not copied when this plugin is created as a library.
// https://forums.swift.org/t/swift-static-libraries-dont-copy-generated-objective-c-header/19816
#import "background_downloader-Swift.h"
#endif

@implementation BackgroundDownloaderPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [Downloader registerWithRegistrar:registrar];
}
@end

File: ios/background_downloader/Sources/background_downloader/BackgroundDownloaderPlugin.swift
//
//  BackgroundDownloaderPlugin.swift
//  background_downloader
//
//  Created by Bram on 1/9/25.
//

import Flutter
import UIKit

@objc(BackgroundDownloaderPlugin) // Keep the Objective-C name for compatibility
public class BackgroundDownloaderPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    BDPlugin.register(with: registrar)
  }
}
